packages/builder/src/components/common/CodeEditor/CodeEditor.svelte

```diff
@@ -139,6 +149,45 @@
     }
   }
 
+  $: promptLoading = false
+  let popoverWidth = 300
+  let suggestedCode: string | null = null
+  let previousContents: string | null = null
+  const generateJs = async (prompt: string) => {
+    previousContents = editor.state.doc.toString()
+    promptLoading = true
+    popoverWidth = 30
+    const { code } = await API.generateJs({ prompt })
+    value = code
+    editor.dispatch({
+      changes: { from: 0, to: editor.state.doc.length, insert: code },
+    })
+    suggestedCode = code
+    popoverWidth = 100
+    promptLoading = false
```

Comment:
Should this be wrapped in a try/finally? Otherwise it will be stuck to loading forever